frontend/components/Campaign.jsx

```diff
@@ -0,0 +1,169 @@
+import React, { Component } from 'react';
+import browserHistory from 'react-router/lib/browserHistory';
+import NotFound from './NotFound.jsx';
+
+import Loading from './Loading.jsx';
+
+import utils from '../utils.js';
+import rootUrl from '../endpoint.js';
+
+
+class DonorRow extends Component {
+    render() {
+        return (
+            <div style={{ fontSize: "0.85rem", marginBottom: "0.5em" }}>
+                <strong style={{ fontWeight: "bold" }}>
+                    { this.props.name }
+                </strong>: { utils.formatCurrency(this.props.amount) }
+            </div>
+        );
+    }
+}
+
+
+export default class Campaign extends Component {
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            large_donors: [],
+            recent_donors: [],
+        }
+
+        this.refresh = this.refresh.bind(this);
+    }
+
+    refresh() {
+        fetch(`${ rootUrl }campaign/${this.props.params.campaign_id}`).then(res => {
+            if (res.ok) {
+                res.json().then(json => {
+                    this.setState({
+                        campaign_id: json.campaign_id,
+                        charity_name: json.charity_name,
+                        secret_id: json.secret_id,
+                        created_on: json.created_on,
+                        notified_on: json.notified_on,
+                        campaigner_email: json.campaigner_email,
+                        campaigner_name: json.campaigner_name,
+                        campaign_status: json.campaign_status,
+                        donation_total_cents: json.donation_total_cents,
+                        match_cents: json.match_cents,
+                        donation_url: json.donation_url,
+                        donation_email: json.donation_email,
+                        charity_id: json.charity_id,
+                        recent_donors: json.recent_donors.map(donor => [donor.donation_cents, donor.donor_name]),
+                        large_donors: json.large_donors.map(donor => [donor.donation_cents, donor.donor_name]),
+                    });
+                })
+            } else {
+                this.setState({ notFound: true })
+            }
+        });
+    }
+
+    componentDidMount() {
+        // TODO: fetch() the campaign here, or redirect to 404.
+        window.setInterval(this.refresh, 30 * 1000);
+        this.refresh();
+    }
+
+    render() {
+        if (this.state.notFound) {
+            return <NotFound />
+        }
+        if (!this.state.donation_url) {
+            return (<Loading />);
+        }
+
+        let title, footerText;
+        if (this.state.campaign_status != "active") {
+            title = this.state.campaigner_name + `'s Campaign (${this.state.campaign_status})`;
```

Comment:
this title doesn't change, just the header-label from "Contribute to" to the campaign's status "cancelled" or "matched"